### PR TITLE
Adding linearmodels recipe

### PR DESCRIPTION
**Pull request by khwilson**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by khwilson** [comment]
Hi all. Sorry for the several rounds of checks here. Took me a while to figure out how to test locally. :-)

This is meant to be a py3k only recipe for the moment. Hopefully I eventually figured out how to do that.

File: recipes/linearmodels/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: 29f07c73be57b5356da41cad83ae53604b215b00cbf3e61347fe5d434cdcfae2
+
+build:
+  number: 0
+  noarch: python
```
Comment (scopatz) [inline recipes/linearmodels/meta.yaml]: I don't think this can be noarch if it relies on Cython

File: recipes/linearmodels/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
```
Comment (scopatz) [inline recipes/linearmodels/meta.yaml]: You probably need a `build` section

File: recipes/linearmodels/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    - numpy >=1.15
+    - pandas >=0.23
+    - patsy
+    - pip >=10
+    - property_cached >=1.6.3
+    - python >=3.4
+    - scipy >=1
+    - statsmodels >=0.9
+  run:
+    - cython >=0.29.14
```
Comment (scopatz) [inline recipes/linearmodels/meta.yaml]: Do you really need cython at runtime?

File: recipes/linearmodels/meta.yaml (hunk 1 above)
Comment (khwilson) [reply recipes/linearmodels/meta.yaml]: This was actually a question I had. Couldn't tell from the docs. The cython files themselves aren't OS specific, but obviously compilation happens at some point. If not noarch, what's the appropriate method of restricting to py3k?

Comment (scopatz) [reply recipes/linearmodels/meta.yaml]: build: skip: True  # [py <30]

File: recipes/linearmodels/meta.yaml (hunk 2 above)
Comment (khwilson) [reply recipes/linearmodels/meta.yaml]: Ah. I have now copied the one from the pandas recipe. Hopefully that is sufficient?

File: recipes/linearmodels/meta.yaml (hunk 3 above)
Comment (khwilson) [reply recipes/linearmodels/meta.yaml]: Probably not. :-) Removed.

File: recipes/linearmodels/meta.yaml (hunk 1 above)
Comment (khwilson) [reply recipes/linearmodels/meta.yaml]: I have copied the pandas recipe and just added a build -> skip. Hopefully that is correct. :-)

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**